win.patch

```diff
@@ -0,0 +1,338 @@
+ worldgen/terrain/source/cpu/elements/voronoi_rocks.cpp    | 2 +-
+ worldgen/terrain/source/cpu/elements/warped_rocks.cpp     | 2 +-
+ worldgen/terrain/source/cpu/elements/waterbody.cpp        | 2 +-
+ worldgen/terrain/source/cpu/meshing/visibility_test.cpp   | 2 +-
+ worldgen/terrain/source/cpu/surfaces/chunkyrock.cpp       | 2 +-
+ worldgen/terrain/source/cpu/surfaces/cobble_stone.cpp     | 2 +-
+ worldgen/terrain/source/cpu/surfaces/cracked_ground.cpp   | 2 +-
+ worldgen/terrain/source/cpu/surfaces/dirt.cpp             | 2 +-
+ worldgen/terrain/source/cpu/surfaces/ice.cpp              | 2 +-
+ worldgen/terrain/source/cpu/surfaces/mountain.cpp         | 2 +-
+ worldgen/terrain/source/cpu/surfaces/mud.cpp              | 2 +-
+ worldgen/terrain/source/cpu/surfaces/sand.cpp             | 2 +-
+ worldgen/terrain/source/cpu/surfaces/sandstone.cpp        | 2 +-
+ worldgen/terrain/source/cpu/surfaces/snow.cpp             | 2 +-
+ worldgen/terrain/source/cpu/surfaces/soil.cpp             | 2 +-
+ worldgen/terrain/source/cpu/surfaces/stone.cpp            | 2 +-
+ worldgen/terrain/utils/ctype_util.py                      | 8 +++++++-
+ 23 files changed, 29 insertions(+), 23 deletions(-)
+
+diff --git a/worldgen/terrain/assets/landtiles/core.py b/worldgen/terrain/assets/landtiles/core.py
+index aa2d075..5360b30 100644
+--- a/worldgen/terrain/assets/landtiles/core.py
++++ b/worldgen/terrain/assets/landtiles/core.py
+@@ -83,7 +83,7 @@ def assets_to_data(
+     N=2048,
+     do_smooth=False,
+ ):
+-    preset_name = str(folder).split("/")[-2]
++    preset_name = str(folder).replace("\\", "/").split("/")[-2]
+     data = {}
+     if land_process is None: path = folder/f"{AssetFile.Heightmap}.exr"
+     elif land_process == Process.Snowfall: path = folder/f"{Process.Snowfall}.{AssetFile.Heightmap}.exr"
+diff --git a/worldgen/terrain/source/cpu/elements/atmosphere.cpp b/worldgen/terrain/source/cpu/elements/atmosphere.cpp
+index b78308d..1a26cb7 100644
+--- a/worldgen/terrain/source/cpu/elements/atmosphere.cpp
++++ b/worldgen/terrain/source/cpu/elements/atmosphere.cpp
+@@ -15,7 +15,7 @@ extern "C" {
+     ) {
+         using namespace data;
+         #pragma omp parallel for
+-        for (size_t idx = 0; idx < size; idx++) {
++        for (int idx = 0; idx < size; idx++) {
+             atmosphere(
+                 positions[idx], sdfs + idx, meta_param,
+                 d_i_params, d_f_params, second_d_i_params, second_d_f_params
+diff --git a/worldgen/terrain/source/cpu/elements/ground.cpp b/worldgen/terrain/source/cpu/elements/ground.cpp
+index 8744d5c..78f2c95 100644
+--- a/worldgen/terrain/source/cpu/elements/ground.cpp
++++ b/worldgen/terrain/source/cpu/elements/ground.cpp
+@@ -19,7 +19,7 @@ extern "C" {
+         int n_auxiliaries = 1;
+         if (auxs == NULL) n_auxiliaries = 0;
+         #pragma omp parallel for
+-        for (size_t idx = 0; idx < size; idx++) {
++        for (int idx = 0; idx < size; idx++) {
```

Comment:
This type change should be made to be a single type that is multiplatform. Is size_t not supported on windows? If so we should use a different type, but it should be some other unsigned int type rather than a signed int.

Reply:
Yes, I received error C3016: 'idx': the index variable in the OpenMP 'for' statement must be a signed integer when compiling with MSVC 19